plugin/supervisor.go

```diff
@@ -17,9 +19,11 @@ import (
 )
 
 type supervisor struct {
-	client      *plugin.Client
-	hooks       Hooks
-	implemented [TotalHooksId]bool
+	client         *plugin.Client
+	clientProtocol plugin.ClientProtocol
```

Comment:
This is used to access the `Ping()` method implemented here:
https://github.com/hashicorp/go-plugin/blob/master/rpc_client.go#L167

The `ClientProtocol` interface is used to be ambiguous to whether the client is a `RPCClient` or `GRPCClient`. There is an [issue in mattermost-server](https://github.com/mattermost/mattermost-server/issues/9486) made to support gRPC. The `ClientProtocol` interface used here may be relevant for going forward with support for gRPC.